### DISCUSSION TITLE
Entregas para promocionar de ejercicios QMP

### DISCUSSION THREAD

**Discussion by AlejandroDeheza** [dds-jv/foro#214] [Consultas generales] [answered]
Buenas!
Entre los del grupo del TP anual nos estamos juntando los domingos a plantear una solución de QMP y hacer un diagrama, luego por separado implementamos la solución. Pero para que todos podamos ir por la promoción ¿Es necesario que cada uno entregue en un repo separado o esta bien que todos entreguemos nuestras soluciones en distintas branches de un mismo repositorio?
Pensábamos poder elegir una de nuestras soluciones para que sepamos que esa es la que finalmente se va a corregir (porque solo se corrige una por grupo). Esa solución la podríamos dejar en el "main" y las demás resoluciones particulares las podríamos dejar en distintas branches del mismo repo.
Mas allá de eso ¿Cuál seria la forma correcta de realizar las entregas para intentar alcanzar la promoción? ¿Estamos haciendo mal al intentar buscar una solución en grupo aunque implementemos por separado?

Gracias!

**Comment by flbulgarelli** [comment, marked as answer]
¡Hola!

> Entre los del grupo del TP anual nos estamos juntando los domingos a plantear una solución de QMP y hacer un diagrama, luego por separado implementamos la solución. 

Perfecto, es una opción posible. De todas formas, mi recomendación (y es sólo eso, una recomendación, no una exigencia) es que algunas entregas las piensen por separado, para ir preparándose al problema de la hoja en blanco individualmente, tal como ocurre en el parcial. 

> Pero para que todos podamos ir por la promoción ¿Es necesario que cada uno entregue en un repo separado o esta bien que todos entreguemos nuestras soluciones en distintas branches de un mismo repositorio?

No, no es necesario. De hecho si hacen lo segundo  y **lo dejan en claro en el README** ayudarán a quien corrija, así sabrá que no tiene que mirar el diagrama de clases N veces. 

> Mas allá de eso ¿Cuál seria la forma correcta de realizar las entregas para intentar alcanzar la promoción? ¿Estamos haciendo mal al intentar buscar una solución en grupo aunque implementemos por separado?

No, mal no está. Como te decía antes, es una posibilidad, y desde el punto de vista de los requisitos, cumple. Pero para que le saquen más provecho, al menos de vez en cuando será beneficio que lo piensen en soledad. 

¡Saludos!

**Reply by AlejandroDeheza** [reply]
Entiendo, muchas gracias!
